### ISSUE TITLE
[Regression] Unable to find provisioning profile generated by match

### ISSUE THREAD

**Issue by a13xb** [fastlane/fastlane#11073]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Xcode 9.1 is struggling to find the ad-hoc provisioning profile managed by `match`. There are a lot of very similar issues floating around, but I couldn't find anything specific to `match`.

Basically an ad-hoc build fails the archive step with an all-too-familiar error:
> Error Domain=IDEProfileLocatorErrorDomain Code=1 "No profiles for 'com.example.app' were found" UserInfo={NSLocalizedDescription=No profiles for 'com.example.app' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS Ad Hoc provisioning profiles matching 'com.example.app'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}

`match` definitely installed the provisioning profile and I don't want to enable `-allowProvisioningUpdates` as most workarounds suggest because that would generate and use Xcode-generated profiles.

What I did next:
1. Clear `~/Library/MobileDevice/Provisioning Profiles` (and derived data folder) and resync. The only provisioning profiles left are the wildcard team profiles and one ad-hoc profile created by `match`.
2. Make sure a single signing identity (used by `match`) is left on the machine (delete the others from the keychain).
3. Check that a correct Xcode installation is used by the command line tools (`xcode-select -p`).

The build is still failing.

### Workaround that worked
I noticed that the following export options were generated:
```
{
  "provisioningProfiles": {
    "com.example.app": "match AdHoc com.example.app"
  },
  "method": "ad-hoc",
  "signingStyle": "automatic"
}
```
1. If I change `signingStyle` from "automatic" to "manual" and rerun the export step, it succeeds.
2. If the `signingStyle` option is removed entirely, the build still fails.

This seems to have been introduced by #11033 (specifically [by changing from "manual" signing style to whatever is set in the project file](https://github.com/fastlane/fastlane/pull/11033/files#diff-1bdc008605f6443a52954ea425ab2430R173)), which is automatic by default. Changing this line back to hardcoded "manual" fixes the build.

But I don't really want to change project settings for this, because I still want automatic management of development certificates.

### One more thing…

I retried the build on another machine at some point, and lo and behold, it worked perfectly. Same version of Xcode/OSX/Fastlane. Just to be sure I cleared out all provisioning profiles again and the build started failing. I inspected the successful build, and it turned out it used a completely wrong provisioning profile, an old Xcode-managed profile, instead of the match profile as specified. So in some circumstances the build may appear to work if you happen some cached valid provisioning profiles that Xcode happens to find.

**TL;DR Xcode ignores the provisioning profile specified in export options, unless signing style is set to "manual".**

### Captured Output

Command Used: `my_custom_lane`
<details><summary>Output/Log</summary>

```

Successfully loaded Appfile at path '/Users/USER/ios/fastlane/Appfile'
- apple_id: 'my@apple.id'
-------
Successfully loaded Appfile at path '/Users/USER/ios/fastlane/Appfile'
- apple_id: 'my@apple.id'
- team_id: 'XXXXXXXXXX'
- app_identifier: 'com.example.app'
-------

+------------+----------------------------------+
|  Detected Values from './fastlane/Matchfile'  |
+------------+----------------------------------+
| username   | my@apple.id                      |
| git_url    | git@github.com:company/match.git |
| git_branch | somebranch                       |
+------------+----------------------------------+


+-----------------------+----------------------------------+
|                 Summary for match 2.67.0                 |
+-----------------------+----------------------------------+
| type                  | adhoc                            |
| force_for_new_devices | true                             |
| git_url               | git@github.com:company/match.git |
| git_branch            | somebranch                       |
| app_identifier        | com.example.app                  |
| username              | my@apple.id                      |
| keychain_name         | login.keychain                   |
| readonly              | false                            |
| team_id               | XXXXXXXXXX                       |
| verbose               | false                            |
| force                 | false                            |
| skip_confirmation     | false                            |
| shallow_clone         | false                            |
| clone_branch_directly | false                            |
| skip_docs             | false                            |
| platform              | ios                              |
+-----------------------+----------------------------------+

Loading session from '/Users/USER/.fastlane/spaceship/my@apple.id/cookie'

+-------------------+-------------------------------------------------------------------------+
|                                    Installed Certificate                                    |
+-------------------+-------------------------------------------------------------------------+
| User ID           | XXXXXXXXXX                                                              |
| Common Name       | iPhone Distribution: ACME, Inc. (XXXXXXXXXX)                            |
| Organisation Unit | XXXXXXXXXX                                                              |
| Organisation      | ACME, Inc.                                                              |
| Country           | US                                                                      |
| Start Datetime    | Feb 26 04:38:57 2017 GMT                                                |
| End Datetime      | Feb 26 04:38:57 2018 GMT                                                |
+-------------------+-------------------------------------------------------------------------+


+---------------------+------------------------+------------------------+
|                    Installed Provisioning Profile                     |
+---------------------+------------------------+------------------------+
| Parameter           | Environment Variable   | Value                  |
+---------------------+------------------------+------------------------+
| App Identifier      |                        | com.example.app        |
| Type                |                        | adhoc                  |
| Platform            |                        | ios                    |
| Profile UUID        | sigh_com.example.app_  | d3381967-bc8e-4c78-a4  |
|                     | adhoc                  | 7a-50413b04ca47        |
| Profile Name        | sigh_com.example.app_  | match AdHoc            |
|                     | adhoc_profile-name     | com.example.app        |
| Profile Path        | sigh_com.example.app_  | /Users/USER/Library/M  |
|                     | adhoc_profile-path     | obileDevice/Provision  |
|                     |                        | ing                    |
|                     |                        | Profiles/d3381967-bc8  |
|                     |                        | e-4c78-a47a-50413b04c  |
|                     |                        | a47.mobileprovision    |
| Development Team ID | sigh_com.example.app_  | XXXXXXXXXX             |
|                     | adhoc_team-id          |                        |
+---------------------+------------------------+------------------------+


+---------------------+---------------------+
| Detected Values from './fastlane/Gymfile' |
+---------------------+---------------------+
| scheme              | CompanyApp          |
+---------------------+---------------------+


+---------------------+---------------------+
| Detected Values from './fastlane/Gymfile' |
+---------------------+---------------------+
| scheme              | CompanyApp          |
+---------------------+---------------------+


+--------------------------------------+--------------------------------------+
|                           Summary for gym 2.67.0                            |
+--------------------------------------+--------------------------------------+
| export_method                        | ad-hoc                               |
| export_options.provisioningProfiles  | match AdHoc com.example.app          |
| .com.example.app                     |                                      |
| workspace                            | ./myapp.xcworkspace                  |
| destination                          | generic/platform=iOS                 |
| output_name                          | CompanyApp                           |
| build_path                           | /Users/USER/Library/Developer/Xcode  |
|                                      | /Archives/2017-11-30                 |
| scheme                               | CompanyApp                           |
| clean                                | false                                |
| output_directory                     | .                                    |
| silent                               | false                                |
| skip_package_ipa                     | false                                |
| buildlog_path                        | ~/Library/Logs/gym                   |
| skip_profile_detection               | false                                |
| xcode_path                           | /Applications/Xcode.app              |
+--------------------------------------+--------------------------------------+

+--------------------------------------+----------------------+
|                   Generated Build Command                   |
+--------------------------------------+----------------------+
| Option                               | Value                |
+--------------------------------------+----------------------+
| set -o pipefail &&                   |                      |
| xcodebuild                           |                      |
| -workspace ./myapp.xcworkspace       |                      |
| -scheme CompanyApp                   |                      |
| -destination                         | generic/platform=iOS |
| -archivePath                         |                      |
| /Users/USER/Library/Developer/Xcode  |                      |
| /Archives/2017-11-30/CompanyApp\     |                      |
| 2017-11-30\ 23.56.50.xcarchive       |                      |
| archive                              |                      |
| | tee                                |                      |
| /Users/USER/Library/Logs/gym/Compan  |                      |
| yApp.log                             |                      |
| | xcpretty                           |                      |
+--------------------------------------+----------------------+
+--------------------------------------+--------------------------------------+
|                          Generated Package Command                          |
+--------------------------------------+--------------------------------------+
| Option                               | Value                                |
+--------------------------------------+--------------------------------------+
| /usr/bin/xcrun                       |                                      |
| /Users/USER/.fastlane/bin/bundle/li  |                                      |
| b/ruby/gems/2.2.0/gems/fastlane-2.6  |                                      |
| 7.0/gym/lib/assets/wrap_xcodebuild/  |                                      |
| xcbuild-safe.sh -exportArchive       |                                      |
| -exportOptionsPlist                  | /var/folders/nr/sfqd96m52q76_f0fl1x  |
|                                      | zr8ch0000gn/T/gym_config20171201-67  |
|                                      | 367-129iq3t.plist                    |
| -archivePath                         |                                      |
| /Users/USER/Library/Developer/Xcode  |                                      |
| /Archives/2017-11-30/CompanyApp\     |                                      |
| 2017-11-30\ 23.56.50.xcarchive       |                                      |
| -exportPath                          | /var/folders/nr/sfqd96m52q76_f0fl1x  |
|                                      | zr8ch0000gn/T/gym_output20171201-67  |
|                                      | 367-rjrezi                           |
+--------------------------------------+--------------------------------------+
+ xcodebuild -exportArchive -exportOptionsPlist /var/folders/nr/sfqd96m52q76_f0fl1xzr8ch0000gn/T/gym_config20171201-67367-129iq3t.plist -archivePath '/Users/USER/Library/Developer/Xcode/Archives/2017-11-30/CompanyApp 2017-11-30 23.56.50.xcarchive' -exportPath /var/folders/nr/sfqd96m52q76_f0fl1xzr8ch0000gn/T/gym_output20171201-67367-rjrezi
2017-12-01 00:00:24.622 xcodebuild[71282:7072084] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/nr/sfqd96m52q76_f0fl1xzr8ch0000gn/T/CompanyApp_2017-12-01_00-00-24.621.xcdistributionlogs'.
2017-12-01 00:00:26.846 xcodebuild[71282:7072084] [MT] IDEDistribution: Step failed: <IDEDistributionSigningAssetsStep: 0x7f999c610a70>: Error Domain=IDEDistributionSigningAssetStepErrorDomain Code=0 "Locating signing assets failed." UserInfo={NSLocalizedDescription=Locating signing assets failed., IDEDistributionSigningAssetStepUnderlyingErrors=(
"Error Domain=IDEProfileLocatorErrorDomain Code=1 \"No profiles for 'com.example.app' were found\" UserInfo={NSLocalizedDescription=No profiles for 'com.example.app' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS Ad Hoc provisioning profiles matching 'com.example.app'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}"
)}
error: exportArchive: No profiles for 'com.example.app' were found

Error Domain=IDEProfileLocatorErrorDomain Code=1 "No profiles for 'com.example.app' were found" UserInfo={NSLocalizedDescription=No profiles for 'com.example.app' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS Ad Hoc provisioning profiles matching 'com.example.app'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.}

** EXPORT FAILED **

+---------------+-------------------------+
|            Build environment            |
+---------------+-------------------------+
| xcode_path    | /Applications/Xcode.app |
| gym_version   | 2.67.0                  |
| export_method | ad-hoc                  |
| sdk           | iPhoneOS11.1.sdk        |
+---------------+-------------------------+


+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | Verifying required     | 0           |
|      | fastlane version       |             |
| 2    | default_platform       | 0           |
| 3    | ensure_git_status_cle  | 0           |
|      | an                     |             |
| 4    | cocoapods              | 13          |
| 5    | Switch to ios beta     | 0           |
|      | lane                   |             |
| 6    | match                  | 18          |
| 💥   | gym                    | 221         |
+------+------------------------+-------------+



```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.15.0                          |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G1036)                  |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.1                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.18.3"
default_platform :ios

platform :ios do
    before_all do
        ensure_git_status_clean
        cocoapods
    end

    private_lane :beta do |options|
        match(type: "adhoc", force_for_new_devices: true)
        gym(export_method: "ad-hoc")
        # ... distribute the build ...
    end

    desc "app2 build"
    lane :company_app2 do |options|
        beta(**options)
    end

    desc "app1 build"
    lane :company_app1 do |options|
        beta(**options)
    end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id ENV["APPLE_ID"]

for_platform :ios do
    for_lane :company_app2 do
        team_id 'YYYYYYYYYY'
        app_identifier 'com.example.app2'
    end
    for_lane :company_app1 do
        team_id 'XXXXXXXXXX'
        app_identifier 'com.example.app'
    end
end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.67.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.3        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.3        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
| psych                     | 2.0.8        |
| libxml-ruby               | 3.0.0        |
</details>

*generated on:* **2017-12-01**
</details>

**Comment by a13xb** [comment]
Thanks, bot, yes, it's always signing. *Always.*

**Comment by alexargo** [comment]
I use cert/sigh and I'm having this same issue. I noticed that before #11033 was pushed to a fastlane release my generated plist looked like this:
```
[22:47:28]: Generated plist file with the following values:
[22:47:28]: ▸ -----------------------------------------
[22:47:28]: ▸ {
[22:47:28]: ▸   "provisioningProfiles": {
[22:47:28]: ▸     "com.astarsoftware.solitaire": "com.astarsoftware.solitaire AppStore",
[22:47:28]: ▸     "7LP83SDNUN.com.astarsoftware.solitaire": "com.astarsoftware.solitaire AppStore"
[22:47:28]: ▸   },
[22:47:28]: ▸   "method": "app-store",
[22:47:28]: ▸   "signingStyle": "manual"
[22:47:28]: ▸ }
[22:47:28]: ▸ -----------------------------------------
```

After updating fastlane from 2.66.2 to 2.67.0... 🚀 I get the following and it fails:

```
[02:10:20]: ▸ -----------------------------------------
[02:10:20]: ▸ {
[02:10:20]: ▸   "signingStyle": "automatic",
[02:10:20]: ▸   "provisioningProfiles": {
[02:10:20]: ▸     "com.astarsoftware.solitaire": "com.astarsoftware.solitaire AppStore",
[02:10:20]: ▸     "7LP83SDNUN.com.astarsoftware.solitaire": "com.astarsoftware.solitaire AppStore"
[02:10:20]: ▸   },
[02:10:20]: ▸   "method": "app-store"
[02:10:20]: ▸ }
[02:10:20]: ▸ -----------------------------------------
```

**Comment by jspohr** [comment]
Just started using fastlane for the first time, and ran into this issue, too. Took me a while to figure out that setting the code signing to manual in the Xcode project works.

**Comment by enahum** [comment]
definitely causing problems, either revert the PR or add a ENV var to bypass it ;)

**Comment by julestburt** [comment]
I successfully change to manual signing during our build process - and use match and ENV var to pass to Gym to build. All works well...but auto never got working, as we create a number of apps (bundleIDs) from the same codebase.

**Comment by jspohr** [comment]
Care to help out a noob and give an example how to use this ENV var to workaround this? Much appreciated!

**Comment by julestburt** [comment]
Hope I can help. Coding is always about moving some data from A to B. In this case it’s a provisioning profile reference, generated by sigh, for your build. I assume you’re using manual code signing too. So, this is about passing the reference to gym which in turn passes this to an xcodebuild command.

The PROVISIONING_PROFILE_SPECIFIER in an ENV variable (environment that you're executing commands in such as terminal/bash) that is set with/as part of the xcodebuild command, just as you would set an ENV variable in a script, which Xcode uses to select the correct provision.

Sigh does in fact install the provision in a place where Xcode can reference it - once it has the uuid.

    udid = sigh(output_path: "./provisions")

    gym(scheme: “MasterProject”,
	export_method: "app-store",
        clean: true,
	xcargs: "PROVISIONING_PROFILE_SPECIFIER='#{udid}'")

If you watch the output you will see sigh and gym execute, then issue and xcodebuild command with the xcargs passed.

Jules.

**Comment by alexargo** [comment]
@a13xb It might be helpful to mark this as [Regression] in the title so it gets more attention.

**Comment by a13xb** [comment]
@alexargo let's see if that helps…

**Comment by milch** [comment]
@alexargo That definitely helped. I've seen this issue around before, but it is very clear that this is a regression introduced in 2.67. 

Sorry everyone that you're having these problems. I'll push out a PR ASAP to revert the setting of the export option. 

@a13xb Thanks for your thorough investigation!
